conformance_tests/sysman/test_sysman_init/src/test_init_sysman.cpp

```diff
@@ -24,4 +244,401 @@ TEST(SysmanInitTests,
   ASSERT_GT(pCount, 0);
 }
 
+TEST(SysmanInitTests, GivenSysmanInitializedThenCallingCoreInitSucceeds) {
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  uint32_t zesInitCount = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+  uint32_t zeInitCount = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDriverGet(&zesInitCount, nullptr));
+  ASSERT_GT(zesInitCount, 0);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDriverGet(&zeInitCount, nullptr));
+  ASSERT_GT(zeInitCount, 0);
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanEnableEnvDisabledAndCoreInitializedFirstWhenSysmanInitializedThenzesDriverGetWorks) {
+  auto is_sysman_enabled = getenv("ZES_ENABLE_SYSMAN");
+  // Disabling enable_sysman env if it's defaultly enabled
+  if (is_sysman_enabled != nullptr && strcmp(is_sysman_enabled, "1") == 0) {
+    char disable_sysman_env[] = "ZES_ENABLE_SYSMAN=0";
+    putenv(disable_sysman_env);
+  }
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+  uint32_t zeInitCount = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  uint32_t zesInitCount = 0;
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeDriverGet(&zeInitCount, nullptr));
+  ASSERT_GT(zeInitCount, 0);
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesDriverGet(&zesInitCount, nullptr));
+  ASSERT_GT(zesInitCount, 0);
+  if (is_sysman_enabled != nullptr && strcmp(is_sysman_enabled, "1") == 0) {
+    char enable_sysman_env[] = "ZES_ENABLE_SYSMAN=1";
+    putenv(enable_sysman_env);
+  }
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanInitializedViaZesInitAndCoreInitializedWithSysmanEnabledWhenEnumFrequencyDomainsIsCalledWithSysmanHandleThenSuccessIsReturned) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<zes_driver_handle_t> zes_drivers = get_zes_drivers();
+  std::vector<zes_device_handle_t> zes_devices =
+      get_zes_devices(zes_drivers[0]);
+  ASSERT_FALSE(zes_devices.empty());
+
+  uint32_t count = 0;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zesDeviceEnumFrequencyDomains(zes_devices[0], &count, nullptr));
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanInitializedViaZesInitAndCoreInitializedWithSysmanEnabledWhenEnumFrequencyDomainsIsCalledWithCoreHandleThenUninitializedErrorIsReturned) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  uint32_t count = 0;
+  EXPECT_EQ(
+      ZE_RESULT_ERROR_UNINITIALIZED,
+      zesDeviceEnumFrequencyDomains(
+          static_cast<zes_device_handle_t>(ze_devices[0]), &count, nullptr));
+}
+
+TEST(
+    SysmanInitTests,
+    GivenValidDeviceWhenBothLegacyInitAndZesInitAreCalledThenOnlyOneOfTheInitSucceeds) {
```

Comment:
Lets avoid LegacyInit and use either zesInit or zeInit

Reply:
updated!